File: agent/internal/client/events.go
package client

import (
	"encoding/json"
	"fmt"
)

type AgentEvent struct {
	AgentID  string `json:"agent_id"`
	Level    string `json:"level"` // "error" | "warning" | "info"
	Message  string `json:"message"`
	Source   string `json:"source"` // "metrics" | "updater" | "service" | ...
}

func (c *Client) SendEvent(ev AgentEvent) error {
	body, _ := json.Marshal(ev)
	resp, err := c.post("/api/v1/agent/events", body)
	if err != nil {
		return err
	}
	resp.Body.Close()
	if resp.StatusCode != 200 && resp.StatusCode != 201 {
		return fmt.Errorf("send event: %d", resp.StatusCode)
	}
	return nil
}

File: src/api/events.ts
import { apiClient } from './client';

export interface AgentEvent {
  id: number;
  timestamp: string;
  agentId: string;
  level: 'error' | 'warning' | 'info';
  source: string;
  message: string;
}

export async function fetchAgentEvents(agentId: string, limit = 50): Promise<AgentEvent[]> {
  const res = await apiClient.get<AgentEvent[]>('/agent/events', {
    params: { agent_id: agentId, limit },
  });
  return res.data;
}
